satpy/readers/li_base_nc.py

```diff
@@ -0,0 +1,769 @@
+            for oc_name in sectors:
+                for var_name in sector_vars:
+                    self.register_dataset(var_name, oc_name)
+
+    def available_datasets(self, configured_datasets=None):
+        """Determine automatically the datasets provided by this file.
+
+        Uses a per product type dataset registration mechanism using the dataset descriptions declared in the reader
+        construction above.
+        """
+        # pass along existing datasets
+        for is_avail, ds_info in (configured_datasets or []):
+            yield is_avail, ds_info
+
+        for ds_info in self.dataset_infos:
+            yield True, ds_info
+
+    def apply_use_rescaling(self, data_array, ds_info=None):
+        """Apply the use_rescaling transform on a given array."""
+        # Here we should apply the rescaling except if it is explicitly requested not to rescale
+        if ds_info is not None and ds_info.get("use_rescaling", True) is not True:
+            return data_array
+
+        # Check if we have the scaling elements:
+        attribs = data_array.attrs
+        if 'scale_factor' in attribs or 'scaling_factor' in attribs or 'add_offset' in attribs:
+            # TODO remove scaling_factor fallback after issue in NetCDF is fixed
+            scale_factor = attribs.setdefault('scale_factor', attribs.get('scaling_factor', 1))
+            add_offset = attribs.setdefault('add_offset', 0)
+
+            data_array = (data_array * scale_factor) + add_offset
+
+            # rescale the valid range accordingly
+            if 'valid_range' in attribs.keys():
+                attribs['valid_range'] = attribs['valid_range'] * scale_factor + add_offset
+
+        data_array.attrs.update(attribs)
+
+        return data_array
+
+    def apply_broadcast_to(self, data_array, ds_info):
+        """Apply the broadcast_to transform on a given array."""
+        ref_var = self.get_transform_reference('broadcast_to', ds_info)
+
+        logger.debug("Broascasting %s to shape %s", ds_info['name'], ref_var.shape)
+        new_array = da.broadcast_to(data_array, ref_var.shape)
+        dims = data_array.dims if data_array.ndim > 0 else ('y',)
+        data_array = xr.DataArray(new_array, coords=data_array.coords, dims=dims, name=data_array.name,
+                                  attrs=data_array.attrs)
+        return data_array
+
+    def apply_accumulate_index_offset(self, data_array, ds_info):
+        """Apply the accumulate_index_offset transform on a given array."""
+        # retrieve the __index_offset here, or create it if missing:
+        # And keep track of the shared ds_info dict to reset it later in combine_info()
+        self.current_ds_info = ds_info
+        offset = ds_info.setdefault('__index_offset', 0)
+
+        ref_var = self.get_transform_reference('accumulate_index_offset', ds_info)
+
+        # Apply the current index_offset already reached on the indices we have in the current dataset:
+        data_array += offset
```

Comment:
In-place operations for `xr.DataArray`s or Dask arrays isn't generally a good idea.
```suggestion
        data_array = data_array + offset
```